lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoder.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader.decoders;
+
+import io.trino.parquet.ParquetEncoding;
+import io.trino.parquet.PrimitiveField;
+import io.trino.parquet.dictionary.Dictionary;
+import io.trino.parquet.reader.SimpleSliceInputStream;
+
+import javax.annotation.Nullable;
+
+public interface ValueDecoder<T>
+{
+    void init(SimpleSliceInputStream input);
+
+    /**
+     * 'values' array needs to be empty, i.e. contain only 0 values.
```

Comment:
This comment seems out of place. There's no array involved in this signature. If anything, it would be an detail of a class that implements this interface and happens to use an array type for T.

Reply:
I've updated the comment here to clarify that it is for `values` which are primitive arrays (which is the most common scenario, BINARY being the only exception).

Reply:
Still, the concept of “0 values” doesn’t apply to every primitive type (e.g, boolean). I think the comment should go in the specific implementations that care about it

Reply:
I've removed this comment now as there wasn't a value decoder which relies on this behaviour in the current PR